ui/pages/settings/security-tab/security-tab.component.js

```diff
@@ -368,17 +375,31 @@ export default class SecurityTab extends PureComponent {
           </div>
         </div>
         <div className="settings-page__content-item">
-          <div className="settings-page__content-item-col">
-            <TextField
-              type="text"
-              value={this.state.ipfsGateway}
-              onChange={(e) => handleIpfsGatewayChange(e.target.value)}
-              error={ipfsGatewayError}
-              fullWidth
-              margin="dense"
-            />
-          </div>
+          <ToggleButton
+            value={this.state.ipfsGateway}
+            onToggle={(value) => {
+              handleIpfsToggle(value);
+              this.setState({ ipfsToggle: Boolean(value) });
```

Comment:
Just a question here: under which case the value is not a Boolean ?

Reply:
the value here is a string, hence I have used Boolean to make sure it returns falsy for an empty string.

Reply:
We have to handle the case where we need to check it based on the input length as well. So, I think this is fine